js/plugins/dotprompt/src/registry.ts

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { PromptAction } from '@genkit-ai/ai';
+import { GenkitError } from '@genkit-ai/core';
+import { lookupAction } from '@genkit-ai/core/registry';
+import { existsSync, readdir, readFileSync } from 'fs';
+import { basename, join, resolve } from 'path';
+import { Dotprompt } from './prompt.js';
+
+export async function lookupPrompt(
+  dir: string,
+  name: string,
+  variant?: string
+): Promise<Dotprompt> {
+  // Expect to find the prompt in the registry
+  const registryPrompt = (await lookupAction(
+    `/prompt/${name}${variant ? `.${variant}` : ''}`
```

Comment:
why not register .prompts under `/prompt/dotprompt/${name}${variant ? `.${variant}` : ''}`? This way `lookupAction` will handle plugin initialization for you.

Reply:
Ok, I wasn't aware of that, but I see how it resolves some issues. I'll try that.

Reply:
This is working now!